### PR TITLE
redirect url to sage_intacct

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

**Review by Sravanksk — APPROVED** [review]

File: src/app/si/expense-groups/expense-groups.component.ts [hunk 1]
```diff
@@ -133,6 +136,29 @@ export class ExpenseGroupsComponent implements OnInit, OnDestroy {
     });
   }
 
+  openInSageIntacct(id) {
```
Comment (ashwin1111) [inline src/app/si/expense-groups/expense-groups.component.ts:139]: : string

File: src/app/si/expense-groups/expense-groups.component.ts [hunk 2]
```diff
@@ -133,6 +136,29 @@ export class ExpenseGroupsComponent implements OnInit, OnDestroy {
...
+  }
+
+  openInSiHandler(clickedExpenseGroup: ExpenseGroup) {
+      // tslint:disable-next-line: deprecation
+      event.preventDefault();
+      // tslint:disable-next-line: deprecation
+      event.stopPropagation();
+      const that = this;
+      that.isLoading = true;
+      that.taskService.getTasksByExpenseGroupId(clickedExpenseGroup.id).subscribe((tasks: Task[]) => {
```
Comment (ashwin1111) [inline src/app/si/expense-groups/expense-groups.component.ts:150]: should be `tasks: Task`, change in service as well.
One expense group id will return one task only, check api code as well just for confirmation.

File: src/app/si/expense-groups/expense-groups.component.ts [hunk 3]
```diff
@@ -133,6 +136,29 @@ export class ExpenseGroupsComponent implements OnInit, OnDestroy {
...
+
+        that.isLoading = false;
+        const completedTask = tasks.filter(task => task.status === 'COMPLETE')[0];
+
+        if (completedTask) {
+          const redirectedId = completedTask.detail.redirected_url_id;
```
Comment (ashwin1111) [inline src/app/si/expense-groups/expense-groups.component.ts:156]: rename according to api

File: src/app/si/expense-groups/expense-groups.component.ts (hunk 2 above)
Comment (NileshPant1999) [reply src/app/si/expense-groups/expense-groups.component.ts:150]: Hey It was returning List with a single object

Comment (ashwin1111) [reply src/app/si/expense-groups/expense-groups.component.ts:150]: Please fix api as well. .get() should do the job.

Comment (NileshPant1999) [reply src/app/si/expense-groups/expense-groups.component.ts:150]: okay 👍🏻

File: src/app/core/models/task.model.ts [hunk 4]
```diff
@@ -1,13 +1,12 @@
...
-import { MappingError } from './mapping-error.model';
 import { SageIntacctError } from './sage-intacct-error.model';
 
 export type Task = {
   bill: number;
   created_at: Date;
-  detail: MappingError[];
-  sage_intacct_errors: SageIntacctError[];
+  detail: any;  // TODO Replace with a suitable model
+  sage_intacct_errors: any; // Todo Replace with a Suitable Model
```
Comment (ashwin1111) [inline src/app/core/models/task.model.ts:9]: why is this changed to any?

File: src/app/si/expense-groups/view-expense-group/group-mapping-error/group-mapping-error.component.ts [hunk 5]
```diff
@@ -27,9 +27,9 @@ export class GroupMappingErrorComponent implements OnInit {
     that.workspaceId = +that.route.snapshot.parent.params.workspace_id;
     that.expenseGroupId = +that.route.snapshot.parent.params.expense_group_id;
     that.isLoading = true;
-    that.taskService.getTasksByExpenseGroupId(that.expenseGroupId).subscribe((res: Task[]) => {
-      that.mappingErrors = new MatTableDataSource(res.map(task => task.detail).reduce((arr1, arr2) => arr1.concat(arr2)));
-      that.count = res[0].detail && res[0].detail.length;
+    that.taskService.getTasksByExpenseGroupId(that.expenseGroupId).subscribe((res: Task) => {
+      that.mappingErrors = new MatTableDataSource(res.detail.reduce((arr1, arr2) => arr1.concat(arr2)));
```
Comment (ashwin1111) [inline src/app/si/expense-groups/view-expense-group/group-mapping-error/group-mapping-error.component.ts:31]: `that.mappingErrors = new MatTableDataSource(res.detail);` should work, please check and change

File: src/app/si/expense-groups/view-expense-group/group-sage-intacct-error/group-sage-intacct-error.component.ts [hunk 6]
```diff
@@ -27,9 +27,9 @@ export class GroupSageIntacctErrorComponent implements OnInit {
     that.workspaceId = +that.route.snapshot.parent.params.workspace_id;
     that.expenseGroupId = +that.route.snapshot.parent.params.expense_group_id;
     that.isLoading = true;
-    that.taskService.getTasksByExpenseGroupId(that.expenseGroupId).subscribe((res: Task[]) => {
-      that.sageIntacctErrors = new MatTableDataSource(res.map(task => task.sage_intacct_errors).reduce((arr1, arr2) => arr1.concat(arr2)));
-      that.count = res[0].sage_intacct_errors && res[0].sage_intacct_errors.length;
+    that.taskService.getTasksByExpenseGroupId(that.expenseGroupId).subscribe((res: Task) => {
+      that.sageIntacctErrors = new MatTableDataSource(res.sage_intacct_errors.reduce((arr1, arr2) => arr1.concat(arr2)));
```
Comment (ashwin1111) [inline src/app/si/expense-groups/view-expense-group/group-sage-intacct-error/group-sage-intacct-error.component.ts:31]: `that.sageIntacctErrors = new MatTableDataSource(res.sage_intacct_errors);` should work, please check and change

File: src/app/core/services/tasks.service.ts [hunk 7]
```diff
@@ -54,7 +54,7 @@ export class TasksService {
     });
   }
 
-  getTasksByExpenseGroupId(expenseGroupId: number): Observable<Task[]> {
+  getTasksByExpenseGroupId(expenseGroupId: number): Observable<Task> {
```
Comment (Shwetabhk) [inline src/app/core/services/tasks.service.ts:57]: Name of the function should be `getTask` instead of `Tasks`

**Review by ashwin1111 — APPROVED** [review]

File: src/app/si/expense-groups/expense-groups.component.ts [hunk 8]
```diff
@@ -133,6 +136,31 @@ export class ExpenseGroupsComponent implements OnInit, OnDestroy {
     });
   }
 
+  openInSageIntacct(id: string) {
+    if(id){
+      this.windowReference.open(`https://www-p02.intacct.com/ia/acct/ur.phtml?.r=${id}`, '_blank');
+    } else {
+      this.windowReference.open(`https://www.intacct.com/ia/acct/login.phtml`, '_blank');
```
Comment (ashwin1111) [inline src/app/si/expense-groups/expense-groups.component.ts:143]: hey no, let's not do it here. Update this in db for old records.

Comment (NileshPant1999) [reply src/app/si/expense-groups/expense-groups.component.ts:143]: nope updating in db will not work

Comment (NileshPant1999) [reply src/app/si/expense-groups/expense-groups.component.ts:143]: if we update in db after login it will through error as we are not storing the whole url

Comment (ashwin1111) [reply src/app/si/expense-groups/expense-groups.component.ts:143]: discussed offline;

Comment (NileshPant1999) [reply src/app/si/expense-groups/expense-groups.component.ts:143]: resolved.
